index.js

```diff
@@ -395,19 +395,27 @@ function checkQueries(queries) {
 }
 
 var cache = {}
+var parseCache = {}
 
 function browserslist(queries, opts) {
   opts = prepareOpts(opts)
   queries = prepareQueries(queries, opts)
   checkQueries(queries)
 
+  var needsPath = parseQueries(queries).some(function (node) {
+    return QUERIES[node.type].needsPath
+  })
   var context = {
     ignoreUnknownVersions: opts.ignoreUnknownVersions,
     dangerousExtend: opts.dangerousExtend,
     mobileToDesktop: opts.mobileToDesktop,
-    path: opts.path,
     env: opts.env
   }
+  Object.defineProperty(context, 'path', needsPath ? { value: opts.path } : { 
```

Comment:
OK, when I finaylly get why we removed `ctx.path` let’s remove it :D But add a comment like

```
// Removing to avoid using ctx.path without marking query
```

Reply:
✅